tools/prow-auto-bumper/pullrequest.go

```diff
@@ -0,0 +1,161 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"github.com/knative/test-infra/shared/ghutil"
+)
+
+func makeCommitSummary(vs versions) string {
+	return fmt.Sprintf("Update prow from %s to %s, and other images as necessary.", vs.oldVersion, vs.newVersion)
+}
+
+func getMatchTitle() string {
+	return "Update prow from"
+}
+
+func generatePRBody(extraMsgs []string) string {
+	var body string
+	if len(extraMsgs) > 0 {
+		body += "Warnings:\n"
+		for _, msg := range extraMsgs {
+			body += fmt.Sprintf("%s\n", msg)
+		}
+		body += "\n"
+	}
+
+	oncaller, err := getOncaller()
+	var assignment string
+	if err == nil {
+		if oncaller != "" {
+			assignment = "/cc @" + oncaller
+		} else {
+			assignment = "Nobody is currently oncall, so falling back to Blunderbuss."
+		}
+	} else {
+		assignment = fmt.Sprintf("An error occurred while finding an assignee: `%s`.\nFalling back to Blunderbuss.", err)
+	}
+
+	return body + assignment
+}
+
+func getOncaller() (string, error) {
+	req, err := http.Get(oncallAddress)
+	if err != nil {
+		return "", err
+	}
+	defer req.Body.Close()
+	if req.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("HTTP error %d (%q) fetching current oncaller", req.StatusCode, req.Status)
+	}
+	oncall := struct {
+		Oncall struct {
+			ToolsInfra string `json:"tools-infra"`
+		} `json:"Oncall"`
+	}{}
+	if err := json.NewDecoder(req.Body).Decode(&oncall); err != nil {
+		return "", err
+	}
+	return oncall.Oncall.ToolsInfra, nil
+}
+
+func makeGitCommit(gi gitInfo, message string, dryrun bool) error {
+	if "" == gi.head {
+		log.Fatal("pushing to empty branch ref is not allowed")
+	}
+	if err := run(
+		"Running 'git add -A'",
+		func() error { return call("git", "add", "-A") },
+		dryrun); err != nil {
+		return fmt.Errorf("failed to git add: %v", err)
+	}
+	commitArgs := []string{"commit", "-m", message}
+	if "" != gi.userName && "" != gi.email {
+		commitArgs = append(commitArgs, "--author", fmt.Sprintf("%s <%s>", gi.userName, gi.email))
+	}
+	if err := run(
+		fmt.Sprintf("Running 'git %s'", strings.Join(commitArgs, " ")),
+		func() error { return call("git", commitArgs...) },
+		dryrun); err != nil {
+		return fmt.Errorf("failed to git commit: %v", err)
+	}
+	pushArgs := []string{"push", "-f", fmt.Sprintf("git@github.com:%s/%s.git", gi.userID, gi.repo),
+		fmt.Sprintf("HEAD:%s", gi.head)}
+	if err := run(
+		fmt.Sprintf("Running 'git %s'", strings.Join(pushArgs, " ")),
+		func() error { return call("git", pushArgs...) },
+		dryrun); err != nil {
```

Comment:
The format looks harder to read, I would suggest
```suggestion
		dryrun,
	); err != nil {
```
The same applies to a couple of other function calls as well.

Reply:
done